### ISSUE TITLE
Parameter decorator inside class - Method should have "self" as first argument (no-self-argument)

### ISSUE THREAD

**Issue by banukk** [PyCQA/pylint#4754]
In the below example of decorator inside class has an additional enclosing decorator for parameters, pylint against the signature return **Method should have "self" as first argument (no-self-argument)** where indeed param can not be a instance

```python
class A:
    **def decor_params(param):**
        def Decorators(func):
            def inner(self):
                print('Decoration started.')
                func(self)
                print('Decoration of function completed.\n')
    
            return inner
    
        return Decorators

@decor_params(param=999)
def fun1(self):
    print('Decorating - Class A methods.')
```

**Comment by Pierre-Sassoulas** [comment]
I don't understand this issue, could you open another one and follow the template for "bug report" please ?
